src/home_audit.ts

```diff
@@ -0,0 +1,807 @@
+{
+    const {leakage_method, leakage_to_outside, duct:ducts} = hvac_distribution
+
+    if(leakage_to_outside && leakage_method === 'qualitative') {
+        addErrorMessage(`systems/hvac/${index}/hvac_distribution/leakage_to_outside`, "Leakage should not be passed for your system if the method is 'qualitative'")
+    }
+
+    // If we have ducts, we need to ensure the fraction is 100%
+    if(ducts) {
+        const total_fraction = getSumOfObjectPropertiesByFieldName(ducts, 'fraction')
+        if(total_fraction !== 1) {
+            ducts.forEach((duct, index) => {
+                if(duct.fraction) {
+                    addErrorMessage(
+                        `systems/hvac/${index}/hvac_distribution/duct/${index}/fraction`,
+                        `Total Duct Fraction must equal 100%`
+                    )
+                }
+            })
+        }
+    }
+}
+
+/**
+ * Check that if the hot water is 'combined' the HVAC system has a boiler
+ */
+function checkHotWaterCategoryValid({category}:HotWater, hvacs:HVACSystem[]): void
+{
+    const hvac_types:string[] = []
+    hvacs.forEach((system) => {
+        const {heating, cooling} = system
+        heating?.type && hvac_types.push(heating.type)
+        cooling?.type && hvac_types.push(cooling.type)
+    })
+    if(!hvac_types.includes('boiler') && category === 'combined') {
+        addErrorMessage(`systems/domestic_hot_water/category`, 'Must have a boiler for combined hot water category')
+    }
+}
+
+/**
+ * Check that fuel is appropriate for the hot water system
+ */
+function checkHotWaterFuelValid({type, fuel_primary}:HotWater): void
+{
+    if(['tankless_coil', 'indirect'].includes(type) && fuel_primary) {
+        addErrorMessage(`systems/domestic_hot_water/fuel_primary`, 'Fuel is only used if type is set to storage or heat pump')
+    } else if(type === 'heat_pump' && fuel_primary !== 'electric') {
+        addErrorMessage(`systems/domestic_hot_water/fuel_primary`, 'Fuel must be electric if type is heat pump')
+    }
+}
+
+/**
+ * Check that efficiency is appropriate for the hot water system
+ */
+function checkHotWaterEfficiencyValid({type, efficiency_method}:HotWater): void
+{
+    if(['heat_pump', 'tankless', 'tankless_coil'].includes(type) && efficiency_method === 'shipment_weighted') {
+        addErrorMessage(`systems/domestic_hot_water/efficiency_method`, 'Invalid Efficiency Method for entered Hot Water Type')
+    }
+}
+
+/**
+ * Check that the year is appropriate for the hot water system
+ */
+function checkHotWaterYearValid({year}:HotWater): void
+{
+    if(year && (1972 > year || (new Date()).getFullYear() < year)) {
+        addErrorMessage(`systems/domestic_hot_water/year`, `Invalid year, must be between 1972 and ${(new Date()).getFullYear()}`)
+    }
+}
+
+/**
+ * Check that the energy factor is valid for the hot water system
+ */
+function checkHotWaterEnergyFactorValid({type, energy_factor}:HotWater): void
```

Comment:
Energy Factor is not a part of the multi-family schema, and it looks like the schema handles this range check anyway, so I think this method can be dropped.

Reply:
See regression_files/DHW_indirect.json

Reply:
There is an issue here that I'm not certain how to resolve - there is a mismatch between the domestic_hot_water definitions in HomeEnergyScore.type.ts and in _base.schema.js. In HomeEnergyScore.type.ts, the fields are:

```
category
type
fuel_primary
efficiency_method
year
energy_factor
```

But in _base.schema.js, which we validate against, the fields are:

```
category
type
fuel_primary
efficiency_method
year
efficiency
efficiency_unit
```

So basically, our type definition says we use energy_factor, but the schema says we use efficiency and efficiency_unit.

There is a note in the type file that it is generated using json-schema-to-typescript. Should I regenerate the type definition with that script? I'm not sure how to do that, though I'm sure I can figure out out.

Reply:
I did remove the check and confirmed by testing that the validation is being handled by the schema

Reply:
@BenjaminStuermer - Did I drop the ball on this?  I remember looking into this and finding the that typescript type had not been correctly built in hes-rest.  The issue required a fix the the schema, which I resolved and was recently approved/merged by NREL.  Do you still need the updated type?

Reply:
@thepriestscottage I don't remember the details, but I definitely updated the type definitions. I think we're good.